server/src/main/java/org/elasticsearch/index/query/IntervalBuilder.java

```diff
@@ -143,38 +143,49 @@ protected static IntervalsSource combineSources(List<IntervalsSource> sources, i
     protected List<IntervalsSource> analyzeTerms(TokenStream ts) throws IOException {
         List<IntervalsSource> terms = new ArrayList<>();
         TermToBytesRefAttribute bytesAtt = ts.addAttribute(TermToBytesRefAttribute.class);
+        PositionIncrementAttribute posAtt = ts.addAttribute(PositionIncrementAttribute.class);
         ts.reset();
         while (ts.incrementToken()) {
             BytesRef term = bytesAtt.getBytesRef();
-            terms.add(Intervals.term(BytesRef.deepCopyOf(term)));
+            int precedingSpaces = posAtt.getPositionIncrement() - 1;
+            terms.add(extend(Intervals.term(BytesRef.deepCopyOf(term)), precedingSpaces));
         }
         ts.end();
         return terms;
     }
 
+    public static IntervalsSource extend(IntervalsSource source, int precedingSpaces) {
+        if (precedingSpaces == 0) {
+            return source;
+        }
+        return Intervals.extend(source, precedingSpaces, 0);
+    }
+
     protected IntervalsSource analyzeSynonyms(TokenStream ts, int maxGaps, boolean ordered) throws IOException {
         List<IntervalsSource> terms = new ArrayList<>();
         List<IntervalsSource> synonyms = new ArrayList<>();
         TermToBytesRefAttribute bytesAtt = ts.addAttribute(TermToBytesRefAttribute.class);
         PositionIncrementAttribute posAtt = ts.addAttribute(PositionIncrementAttribute.class);
         ts.reset();
         while (ts.incrementToken()) {
-            if (posAtt.getPositionIncrement() == 1) {
+            int spaces = posAtt.getPositionIncrement() - 1;
+            if (spaces >= 0) {
                 if (synonyms.size() == 1) {
-                    terms.add(synonyms.get(0));
+                    terms.add(extend(synonyms.get(0), spaces));
```

Comment:
The `synonyms` contains the word at the position **before** the gap so the `extend` should be applied forward ?

Reply:
It applies backwards because we're using PositionIncrement to detect when there's a gap preceding us in the TokenStream.  If you've got a posInc of 2, that means theres a gap before you in the stream, so you need to extend backwards 1 to cover it.

Reply:
Ah, actually no, it's a bug - I added the test you suggested and it fails, have updated.